### DISCUSSION TITLE
Tournament client: adding "protect" mode to map pool screen

### DISCUSSION THREAD

**Discussion by bdach** [ppy/osu#24336] [Ideas]
Follow up from [discord discussion](https://discord.com/channels/188630481301012481/188630652340404224/1132638423065296908).

osu!mania 4K world cup 2023 ruleset seems to employ a concept of a [protected map](https://osu.ppy.sh/wiki/en/Tournaments/MWC/2023_4K#match-instructions). As far as I gather the meaning of such a designation is as follows:

- A protected map can no longer be banned by either team
- A protect differs from a pick in that a protected map doesn't have to get picked, and in the pick phase protects stop mattering in any way (i.e. final gameplay order is determined by order of picks alone)

which is a bit difficult to visually convey. The solutions proposed on discord include: fading out the protected item in a ban phase, and/or adding a padlock icon to the item (potentially coloured to indicate which team protected it).

Opening for discussion on whether this should be supported in tournament client, and if yes, how it should look.

It was indicated that it would be nice to have this feature in by the qualifier stage which is scheduled to begin [on August 12](https://osu.ppy.sh/wiki/en/Tournaments/MWC/2023_4K#tournament-schedule).

**Comment by cdwcgt** [comment]
Here is my current design

before pick start:
![image](https://github.com/ppy/osu/assets/34775378/b1faa179-03e2-48b2-a9dd-2af6a7fe464e)

after pick start:
![image](https://github.com/ppy/osu/assets/34775378/c7452a2f-69f3-475e-b619-9ba70208db98)

Please let me know if you have any suggestions, I'm not sure if it is right

**Reply by bdach** [reply]
To begin with, the shield thing shouldn't make the text in front of it (behind it?) unreadable.

Maybe it could occupy a corner or something. I dunno.

**Reply by cdwcgt** [reply]
I'm still confused about what background color and position it should use.
This high probability will be affected by the beatmap cover itself
but the icon is indeed at the bottom of the text, because the white background makes it difficult to read(Sorry for choosing an inappropriate color)

**Comment by cdwcgt** [comment]
https://github.com/ppy/osu/assets/34775378/e5dc680c-572c-4902-b6ff-355995be591d

A effect my friend gave me, he don't know how to use github so let me send it for him.
Because currently lazer (looks like?) does not have a dotted line implementation.
Designed to use triangles and tint to act as a dynamic frame.

He also mentions that you can use an image like this:
![image](https://github.com/ppy/osu/assets/34775378/b3f6f456-28c6-4f3a-ab47-cb0a6a329056)
implement dynamic dotted borders by tiling and moving down

**Comment by RushFTK** [comment]
In my opinion, which is also main thread said. major challenge in "protect" visual effect is:
1. how to show "protect" much important on ban phase but less important on pick phase?
2. An team can pick opposite's "protect" map (even it's rare), so how to show picks and protects individual.
the second solution on above might confused on point 2, we may aware about that.

**Reply by RushFTK** [reply]
Also i think in first solution, the shield block text is something acceptable since it's won't prevent audience from watching too much times.Also i think in first solution, the shield block text is something acceptable since it's won't prevent audience from watching too much times.

**Comment by RushFTK** [comment]
anyway above soultion are something "could be achieved". we still waiting UI designer to get "best practice".

**Comment by peppy** [comment]
No discussion is really required, design should be provided by ask rather than crowd-sourced in a convoluted github discussion.

Add a circle in the top right corner with a shield in it, background colour of the circle is the team that protected it. Should only display when enabled.
